File: 20A40/cir-Israel/Ejercicio 25/Referencia.py
class Referencia:   # traducido de Eckel por CST
  def __init__(self,x): self.x = x

  def incremento(self) :
    self.x += 1
    return self
      
  def main(self) :
    r = Referencia(0)
    r.incremento().incremento().incremento()
    print("\nx= ", r.x, "\n")

m = Referencia(0)
m.main()

File: 20A40/cir-Israel/Ejercicio 26/Referencia.cpp
#include<iostream>

using namespace std; // traducido por CST

class referencia
{  private: int x;
   public:
      referencia() {}
      referencia &incremento()
      {  x++;
         return *this;
      }
      void print() { cout << "x = " << x << endl; }
};

           // Chained function calls. All calls
int main() //  modify the same object as same object
{  referencia r = referencia();  // is returned

   r.incremento().incremento().incremento();
   r.print();
   return 0;
}